src/insns/acperm_32bit.adoc

```diff
@@ -29,41 +29,39 @@ ACPERM performs the following operations:
-clear all AP permissions and the <<m_bit>>, and skip the next step
+clear all AP permissions, and the <<m_bit>> to {CAP_MODE_VALUE}. Skip the next step.
+.. If <<cheri_levels_ext,{cheri_levels_ext_name}>> is implemented then also set <<section_cap_level,CL>> to 0 (_local_).
 . Clear AP permissions as required to meet the rules below.
-. Encode the AP permissions for MXLEN=32 according to <<cap_perms_encoding32>>.
+. Encode the AP permissions for RV32 according to <<cap_perms_encoding32>>.
 . Copy `cs1` to `cd`, and update the AP and SDP fields with the newly calculated versions.
 . Set `cd.tag=0` if `cs1` is sealed or if any reserved fields of `cs1` are set.
-+
-Some combinations of permissions cannot be encoded for MXLEN=32, and are not useful when MXLEN=64.
-These cases are defined to return useful minimal sets of permissions, which may be no permissions.
-+
-NOTE: Future extensions may allow more combinations of permissions, especially for MXLEN=64.
-
-The common rules are:
-
-. <<asr_perm>> cannot be set without <<x_perm>> being set
-.. Clear <<asr_perm>> unless <<x_perm>> is set
-. <<c_perm>> cannot be set without at least one of <<r_perm>> or <<w_perm>> being set.
-.. Clear <<c_perm>> unless <<r_perm>> or <<w_perm>> are set.
-. <<m_bit>> cannot be set without <<x_perm>> being set
-.. Clear <<m_bit>> unless <<x_perm>> is set
-. <<lm_perm>> cannot be set without <<c_perm>> being set
-.. Clear <<lm_perm>> unless <<c_perm>> is set.
-. <<sl_perm>> cannot be set without <<c_perm>> being set
-.. Zero <<sl_perm>> unless <<c_perm>> is set.
-. <<el_perm>> cannot be set without <<c_perm>> being set
-.. Zero <<el_perm>> unless <<c_perm>> is set.
-
-NOTE: The combination of <<x_perm>> clear and <<m_bit>> set is reserved for future extensions.
-
-The MXLEN=32 additional rules are:
-
-. Clear <<asr_perm>> unless _all_ other permissions are set
-. Clear <<c_perm>> and <<x_perm>> if <<r_perm>> is not set
-. Clear <<w_perm>> if <<c_perm>> _is_ set, but <<lm_perm>> _is not_ set.
-. Clear <<x_perm>> if <<x_perm>> and <<r_perm>> are set, and <<c_perm>> and <<w_perm>> are both clear.
-. If removing <<lm_perm>> results in unrepresentable permissions, then set <<r_perm>> and <<c_perm>> only.
+
+The rules from <<acperm_rules>> must be followed when removing permissions.
+
+[#acperm_rules]
+.ACPERM common rules
+[%autowidth,float="center",align="center",cols="3,3,3",options="header"]
+|===
+| Rule            | Permission   | Only valid if
+|  1  (RV32 only) | <<asr_perm>> | All other permissions are set.
+|  2              | <<c_perm>>   | <<r_perm>> or <<w_perm>>
+|  3  (RV32 only) | <<c_perm>>   | <<r_perm>>
+|  4  (RV32 only) | <<x_perm>>   | <<r_perm>>
+|  5  (RV32 only) | <<w_perm>>   | not(<<c_perm>>) or (<<c_perm>> and not(<<lm_perm>>))
+|  6  (RV32 only) | <<x_perm>>   | <<w_perm>> or <<c_perm>>
+|  7  (RV32 only) | <<x_perm>>   | <<c_perm>> == <<lm_perm>> == <<el_perm>>
+|  8              | <<sl_perm>>  | <<w_perm>> and <<c_perm>>
```

Comment:
Still very much of the opinion that it was correct above and this was only conditioned on C, not W,  so that a read-only cap could still have a store-level for the purpose of future extensions where we limit the store level transtively.

Reply:
I meant the text before change "as above" (just because thats where it appeared on my screen...)


It was

```
. <<sl_perm>> cannot be set without <<c_perm>> being set
.. Zero <<sl_perm>> unless <<c_perm>> is set.
```

Which had no condition on W. This is not to protect a transition, but to ensure if we introduce a future implict ANDPERM with SL on load, then even capabilities without W can theoretically have an SL so that such an AND does what we want.